work_queue/src/work_queue_process.c

```diff
@@ -29,21 +30,44 @@
 #define TMP_SCRIPT "tmp.sh"
 #define DEFAULT_EXE_APP "#!/bin/sh"
 
-struct work_queue_process *work_queue_process_create(int taskid)
+struct work_queue_process *work_queue_process_create(struct work_queue_task *wq_task)
 {
 	struct work_queue_process *p = malloc(sizeof(*p));
 	memset(p, 0, sizeof(*p));
-	p->task = work_queue_task_create(0);
-	p->task->taskid = taskid;
+	//p->task = work_queue_task_create(0);
+	p->task = wq_task;
+	int taskid = (int) p->task->taskid;
```

Comment:
Casting is usually a danger sign!  If p->task->taskid is something other than an int, then use the same type for taskid.